pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -68,6 +68,10 @@ lib.makeOverridable ({ # The kernel source tarball.
 , preferBuiltin ? stdenv.hostPlatform.linux-kernel.preferBuiltin or false
 , kernelArch ? stdenv.hostPlatform.linuxArch
 , kernelTests ? []
+
+, stdenv ? args'.stdenv
```

Comment:
Do we really want to have this complexity just for ancient hardened kernels? Can we drop them?

Reply:
For context, the only reason we even need this is because hardened kernels rely on GCC plugins, which puts constraints on the GCC version beyond the normal kernel requirements.